Lib/_pyrepl/simple_interact.py

```diff
@@ -82,17 +82,15 @@ def _clear_screen():
 
 
 def run_multiline_interactive_console(
-    namespace: dict[str, Any],
+    *,
+    console: code.InteractiveConsole,
     future_flags: int = 0,
-    console: code.InteractiveConsole | None = None,
 ) -> None:
     from .readline import _setup
+
+    namespace = console.locals if isinstance(console.locals, dict) else dict(console.locals)
```

Comment:
Hmm, does this need the extra `isinstance` check? AFAIK, `console.locals` will always be a dict, or at least a mapping. Please correct me if I'm wrong!

Reply:
I think Milan tried to appease Mypy here. I have a better way.

Reply:
Sigh, actually RLCompleter specifically checks for a `dict` instance:
https://github.com/python/cpython/blob/82a4dac9f6131954c32dac9d0277283fc5b499a9/Lib/rlcompleter.py#L58